src/main/resources/templates/db_scripts/schema.json

```diff
@@ -19,29 +19,79 @@
   "tables": [
     {
       "tableName": "alert",
-      "generateId": true,
       "fromModuleVersion": 1.0,
       "withMetadata": true,
       "populateJsonWithId": true
     },
     {
       "tableName": "reporting_code",
-      "generateId": true,
       "fromModuleVersion": 1.0,
       "withMetadata": true,
       "populateJsonWithId": true
     },
     {
       "tableName": "purchase_order",
-      "generateId": true,
       "fromModuleVersion": 1.0,
       "withMetadata": true,
       "populateJsonWithId": true,
-      "customSnippetPath": "purchase_order_table.sql"
+      "customSnippetPath": "purchase_order_table.sql",
+      "index": [
+        {
+          "fieldName": "assignedTo",
+          "caseSensitive": false
+        },
+        {
+          "fieldName": "metadata.createdDate",
+          "caseSensitive": false
+        },
+        {
+          "fieldName": "metadata.createdByUserId",
+          "caseSensitive": false
+        },
+        {
+          "fieldName": "workflowStatus",
+          "caseSensitive": false
+        },
+        {
+          "fieldName": "approved",
+          "caseSensitive": false
+        },
+        {
+          "fieldName": "dateOrdered",
+          "caseSensitive": false
+        },
+        {
+          "fieldName": "orderType",
+          "caseSensitive": false
+        },
+        {
+          "fieldName": "reEncumber",
+          "caseSensitive": false
+        },
+        {
+          "fieldName": "renewal.manualRenewal",
+          "caseSensitive": false
+        },
+        {
+          "fieldName": "renewal.renewalDate",
+          "caseSensitive": false
+        },
+        {
+          "fieldName": "renewal.reviewPeriod",
+          "caseSensitive": false
+        }
+      ],
+      "fullTextIndex": [
+        {
+          "fieldName": "closeReason.reason"
+        },
+        {
+          "fieldName": "closeReason.note"
+        }
+      ]
```

Comment:
would also make the UI searched better if a ginIndex is created on poNumber, where wildcard searches are used

Reply:
I'm not sure that we need this for poNumber. I have discussed with @aliaksei-chumakou this matter - UI doesn't supports wildcard search on poNumber at the moment. Moreover, unique index already created for poNumber in the schema. So, could you please confirm that this makes sense now?

Reply:
@KVupp we use strict searches from ui on po and po line numbers. So it's more a question to POs if they need wildcard searches.

Reply:
<img width="1667" alt="Screen Shot 2019-06-21 at 7 39 31 AM" src="https://user-images.githubusercontent.com/41596468/59920219-cf8f7000-93f7-11e9-9461-9cc9b41db8bc.png">

I see a query like "*0" working.
 Apart from all that, any query in the search bar forms a query like the below:
<img width="1551" alt="Screen Shot 2019-06-21 at 7 41 55 AM" src="https://user-images.githubusercontent.com/41596468/59920294-154c3880-93f8-11e9-81be-6c4b4f5660c8.png">

I will have to look at how CQL converts these queries to SQL queries, but I would assume they would be regex queries

Reply:
@KVupp We don't support such search query from UI. It's a security issue, as for me, when user input isn't escaped and goes directly to database query

Reply:
https://xkcd.com/327/

Reply:
Also from [RMB](https://github.com/folio-org/raml-module-builder#important-information)
> Behind the scenes, the CQL to Postgres query converter will generate regex queries for = >queries. For example: ?query=fieldA=ABC will generate an SQL regex query, which will require >a gin index to perform on large tables.

so from the CQL queries from the UI, it looks like we are always using "=" , which would generate regex queries on SQL, so I think it would still makes sense to create ginIndexes

Reply:
@KVupp Does it make sense to use `=` on UI for strict equal? Maybe it's better to fix UI to use `==` ?

Reply:
@aliaksei-chumakou ,Like you said I would prefer the POs answer this question, because it comes down to what kind of searches we want to support. 
Atleast on inventory app, these same searches are supported and queries are formed in the similar manner

Reply:
We should definitely get some input from the POs, but my guess is that users are going to want/expect partial matches w/o explicitly using wildcards; at least for some searches...  so if they type in `52` in the search box for a po number search, they'd get results like 10052, 10520, 15200, etc.  

I would have thought that the UI would achieve this by adding (escaped if needed) wildcards to both ends of the search criteria, and using an `=`; e.g. `poNumber=\*52\*`

Reply:
hmm doesn't look like you need to escape the asterisks...

```$ curl $OKAPI/orders/composite-orders --data-urlencode 'query=poNumber=*17*' -H "X-Okapi-Token: $TOKEN" -w '\n' -G
{
  "purchaseOrders" : [ {
    "id" : "4d674341-c40b-4e1c-a993-9d6e8372abd3",
    "approved" : false,
    "dateOrdered" : "2019-06-21T13:06:02.762+0000",
    "notes" : [ "Open Order for Negative API Tests" ],
    "poNumber" : "10170",
    "orderType" : "One-Time",
    "reEncumber" : false,
    "vendor" : "197c79c2-993c-4130-8ae3-a3a8899794d3",
    "workflowStatus" : "Open",
    "metadata" : {
      "createdDate" : "2019-06-21T13:06:02.741+0000",
      "createdByUserId" : "00000001-1111-5555-9999-999999999999",
      "updatedDate" : "2019-06-21T13:06:02.792+0000",
      "updatedByUserId" : "00000001-1111-5555-9999-999999999999"
    }
  }, {
    "id" : "27b3359d-e92e-43f5-b5ff-0fa46bd0bb3a",
    "approved" : false,
    "notes" : [ "Closed Order for Negative API Tests" ],
    "poNumber" : "10171",
    "orderType" : "One-Time",
    "reEncumber" : false,
    "vendor" : "197c79c2-993c-4130-8ae3-a3a8899794d3",
    "workflowStatus" : "Closed",
    "metadata" : {
      "createdDate" : "2019-06-21T13:06:05.585+0000",
      "createdByUserId" : "00000001-1111-5555-9999-999999999999",
      "updatedDate" : "2019-06-21T13:06:05.585+0000",
      "updatedByUserId" : "00000001-1111-5555-9999-999999999999"
    }
  }, {
    "id" : "a727f69c-6e40-40fe-906e-6ea64df78f42",
    "approved" : false,
    "notes" : [ "Pending Order for Negative API Tests" ],
    "poNumber" : "10172",
    "orderType" : "One-Time",
    "reEncumber" : false,
    "vendor" : "197c79c2-993c-4130-8ae3-a3a8899794d3",
    "workflowStatus" : "Pending",
    "metadata" : {
      "createdDate" : "2019-06-21T13:06:07.975+0000",
      "createdByUserId" : "00000001-1111-5555-9999-999999999999",
      "updatedDate" : "2019-06-21T13:06:07.975+0000",
      "updatedByUserId" : "00000001-1111-5555-9999-999999999999"
    }
  }, {
    "id" : "cdb6d204-e12d-43e3-b7dd-fd4acf0f89fc",
    "approved" : true,
    "assignedTo" : "ab18897b-0e40-4f31-896b-9c9adc979a88",
    "notes" : [ "ABCDEFGHIJKLMNO", "ABCDEFGHIJKLMNOPQRST", "ABCDEFGHIJKLMNOPQRSTUV" ],
    "poNumber" : "10174",
    "orderType" : "One-Time",
    "reEncumber" : false,
    "vendor" : "8bb4efff-e6db-446e-8708-4cd14e5625dc",
    "workflowStatus" : "Pending",
    "metadata" : {
      "createdDate" : "2019-06-21T13:08:48.391+0000",
      "createdByUserId" : "00000001-1111-5555-9999-999999999999",
      "updatedDate" : "2019-06-21T13:08:48.391+0000",
      "updatedByUserId" : "00000001-1111-5555-9999-999999999999"
    }
  } ],
  "totalRecords" : 4
}
```

Reply:
OK, 
so let it be ginIndex for poNumber/poLineNumber at least for now.
Thanks for clarification!

Reply:
Leave as-is for now.  If it turns out that A) these searches are required and B) are too slow, we can revisit this.

Reply:
I suspect that the (unique) btree index will provide sufficient performance, even with these types of queries.  If supported, a ginindex would likely provide even greater performance, but postgres apparently doesn't support unique ginindex (thanks for the link @khandramai).